src/pages/permission-management/role/list/Index.js

```diff
@@ -0,0 +1,204 @@
+import React, { PureComponent, Fragment } from 'react';
+import router from 'umi/router';
+import { connect } from 'dva';
+import {
+  List,
+  Card,
+  Row,
+  Col,
+  Radio,
+  Input,
+  InputNumber,
+  Progress,
+  Button,
+  Icon,
+  Dropdown,
+  Menu,
+  Avatar,
+  Modal,
+  Form,
+  DatePicker,
+  Select,
+  Badge,
+  Popconfirm,
+  message,
+  Divider,
+} from 'antd';
+import StandardTable from '@/components/StandardTable';
+import PageHeaderWrapper from '@/components/PageHeaderWrapper';
+import styles from '../../../List/TableList.less';
+const { MonthPicker, RangePicker, WeekPicker } = DatePicker;
+const FormItem = Form.Item;
+
+@connect(({ role, loading }) => ({
+  role,
+  loading: loading.models.role,
+}))
+@Form.create()
+class RoleList extends PureComponent {
+  state = {
+    selectedRows: [],
+    expandForm: false,
+  };
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'role/fetch',
+      payload: {},
+    });
+  }
+
+  handleStandardTableChange = (pagination, filtersArg, sorter) => {
+    const { dispatch } = this.props;
+    const { formValues } = this.state;
+
+    const filters = Object.keys(filtersArg).reduce((obj, key) => {
+      const newObj = { ...obj };
+      newObj[key] = getValue(filtersArg[key]);
+      return newObj;
+    }, {});
+
+    const params = {
+      currentPage: pagination.current,
+      pageSize: pagination.pageSize,
+      ...formValues,
+      ...filters,
+    };
+    if (sorter.field) {
+      params.sorter = `${sorter.field}_${sorter.order}`;
+    }
+
+    dispatch({
+      type: 'role/fetch',
+      payload: params,
+    });
+  };
+
+  handleSelectRows = rows => {
+    this.setState({
+      selectedRows: rows,
+    });
+  };
+
+  handleAdd = () => {
+    router.push('/permission-management/role/add');
+  };
+
+  handleEdit = record => {
+    router.push('/permission-management/role/edit/' + record.id);
+  };
+
+  handleDelete = record => {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'role/delete',
+      payload: record.id,
+    });
+  };
+
+  handleMenuClick = e => {
+    console.log(e);
+    const { selectedRows } = this.state;
+    if (selectedRows.length === 0) return;
+    switch (e.key) {
+      case 'permissAuth':
+        if (selectedRows.length === 1) {
+          router.push('/permission-management/role/permissAuth/' + selectedRows[0].id);
+        } else {
+          message.error('请选择一个角色进行授权!');
+        }
+        break;
+      default:
+        break;
+    }
+  };
+  columns = [
```

Comment:
Codacy found an issue: [columns should be placed before componentDidMount](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=6006324)